### ISSUE TITLE
Move `serializableFields` out into the generator.

### ISSUE THREAD

**Issue by back2dos** [haxetink/tink_typecrawler#12]
Just making an issue here because this should be done in concert with updating the dependencies. This might also be a good time to move the driving functionality into the generator too.

**Comment by kevinresol** [comment]
Could you elaborate?

**Comment by back2dos** [comment]
Right now [`serializableFields`](https://github.com/haxetink/tink_typecrawler/blob/master/src/tink/typecrawler/Crawler.hx#L179) is a hard-coded function in the crawler. The part that converts `ClassField` to `FieldInfo` is fine, but the bit that determines which fields to include should be a method on the generator and the crawler should call it. Something like `function shouldIncludeField(c:ClassField):Bool` or `function shouldIncludeField(c:ClassField, owner:Option<ClassType>):Bool` if you see us dealing with classes any time soon.

Also, there's the `drive` function I just added that gives some control over the generation process (this was mostly so that we can have special treatment of `tink.json.Value`). I think maybe it should be a method on the generator itself, rather than being an extra function - which I chose to go with mostly just to avoid breaking dependencies.
